### ISSUE TITLE
Scores are VERY LOW ? Any reason why ?

### ISSUE THREAD

**Issue by thusinh1969** [ZFTurbo/Keras-RetinaNet-for-Open-Images-Challenge-2018#25]
I use standard codes from your git. Load BGR fast - preprocess input - resize - predict on batch - rescale... !

from keras_retinanet.utils.image import read_image_bgr, preprocess_image, resize_image
from keras_retinanet import models
model_path = './V2_2019/retinanet_resnet50_500_classes_0.4594_converted.h5'
backbone = 'resnet50'
with tf.device('/gpu:0'): 
    model  = models.load_model(model_path, backbone_name=backbone)
    model.compile()
    model.trainable = False

# Just a desk with a computer on top of it
img_original = cv2.imread('./samples/game_desk.jpg')
image = preprocess_image(img_original)
if backbone == 'resnet152':
    image, scale = resize_image(image, min_side=600, max_side=800)
elif backbone == 'resnet101':
    image, scale = resize_image(image, min_side=768, max_side=1024)
else:
    image, scale = resize_image(image, min_side=768, max_side=1024)
image = np.expand_dims(image, 0)
boxes, scores, labels = model.predict_on_batch(image)
boxes /= scale

# Score VERY low < 0.02
print (scores)

array([0.08214764, 0.06412603, 0.05601705, 0.04571734, 0.04532317,
       0.04468972, 0.04217904, 0.0407993 , 0.04019322, 0.03925166,
       0.03771246, 0.03559722, 0.03550753, 0.03535734, 0.03532328,
       0.0347053 , 0.03449641, 0.03443889, 0.03411521, 0.03362035,
       0.033617  , 0.03343837, 0.03295262, 0.03294588, 0.03240565,
       0.03233369, 0.03231934, 0.03231192, 0.03228894, 0.03184201,
       0.03163914, 0.03142316, 0.03140194, 0.0305719 , 0.03034266,
       0.03030455, 0.03029803, 0.03000976, 0.02984777, 0.02961908,
       0.02949041, 0.02917261, 0.02893128, 0.02877107, 0.02872788,
       0.02867853, 0.02866829, 0.0285997 , 0.02834851, 0.02830449,
       0.02773562, 0.02763576, 0.02745436, 0.02736399, 0.02731292,
       0.02712597, 0.02694507, 0.02694022, 0.02692809, 0.02675896,
       0.02671816, 0.02668723, 0.02654719, 0.02654446, 0.02651278,
       0.0264549 , 0.026398  , 0.02639764, 0.02601669, 0.02560534,
       0.0253144 , 0.02520229, 0.02512381, 0.02475631, 0.0247244 ,
       0.02455944, 0.02450828, 0.02446357, 0.02437781, 0.02436773,
       0.02435246, 0.02424899, 0.02402328, 0.02395525, 0.02381209,
       0.02377224, 0.02369539, 0.02356794, 0.0235479 , 0.0233748 ,
       0.02336434, 0.02326049, 0.02324282, 0.02312373, 0.02303222,
       0.02300981, 0.02295618, 0.02285378, 0.02283458, 0.02282603,
       0.02278564, 0.02276726, 0.022741  , 0.02266823, 0.02264441,
       0.02259097, 0.02254979, 0.02235727, 0.02219211, 0.02204609,
       0.02202853, 0.02202193, 0.02199629, 0.02196247, 0.02188792,
       0.02186287, 0.02176195, 0.02174086, 0.0217198 , 0.02163081,
       0.02161748, 0.0215974 , 0.02155071, 0.02153325, 0.02152195,
       0.02146714, 0.02137102, 0.02122512, 0.02118728, 0.02110581,
       0.02107801, 0.02107615, 0.02100795, 0.02091308, 0.02087946,
       0.02075831, 0.02068732, 0.02068458, 0.0206733 , 0.02065487,
       0.02064842, 0.02061027, 0.02059928, 0.02053488, 0.0205096 ,
       0.02050141, 0.02038281, 0.02028204, 0.02025107, 0.02013932,
       0.02012402, 0.02007895, 0.02002274, 0.01998274, 0.01993219,
       0.01991616, 0.01991414, 0.01989718, 0.01984977, 0.01970827,
       0.01965398, 0.01959964, 0.019573  , 0.01955963, 0.01952934,
       0.01952427, 0.01947998, 0.01947166, 0.01943158, 0.01938101,
       0.0193553 , 0.0193505 , 0.0193419 , 0.01926245, 0.01925097,
       0.01923176, 0.01919156, 0.0191825 , 0.01916005, 0.01915716,
       0.01915331, 0.01907359, 0.01903782, 0.01903422, 0.01886127,
       0.01882554, 0.01882288, 0.01880884, 0.01879298, 0.01877416,
       0.0187136 , 0.01869194, 0.01864838, 0.01862114, 0.01856415,
       0.01852076, 0.01851707, 0.01841055, 0.01830504, 0.01828685,
       0.01826145, 0.01808002, 0.01807458, 0.01805865, 0.01804617,
       0.01802996, 0.01799886, 0.01799701, 0.01798505, 0.01796073,
       0.01791422, 0.01787511, 0.01786785, 0.01783917, 0.01782647,
       0.01782295, 0.017804  , 0.01779966, 0.01776807, 0.01773656,
       0.01773415, 0.01771753, 0.01765338, 0.01763948, 0.01763309,
       0.01754763, 0.01745315, 0.01745139, 0.01744092, 0.01737041,
       0.01735855, 0.01733752, 0.01732908, 0.01731344, 0.01730322,
       0.01728839, 0.01728693, 0.01728606, 0.01725296, 0.01721361,
       0.01719457, 0.01715928, 0.01714384, 0.01714236, 0.0171222 ,
       0.01710992, 0.01710517, 0.01710477, 0.01710331, 0.01703121,
       0.01702078, 0.01698007, 0.01697412, 0.01691954, 0.01690822,
       0.01690313, 0.01689106, 0.01688855, 0.01688655, 0.0168758 ,
       0.01687372, 0.01686491, 0.01683687, 0.01682049, 0.01680447,
       0.01675804, 0.01673518, 0.016731  , 0.01670613, 0.01670535,
       0.01668603, 0.0166858 , 0.016675  , 0.01664689, 0.01664315,
       0.01664168, 0.01663797, 0.01660788, 0.01651526, 0.01648847,
       0.01648645, 0.01647347, 0.0164711 , 0.01645983, 0.01645834,
       0.01639759, 0.01637122, 0.01635642, 0.01634395, 0.01631755,
       0.01631425, 0.01630697, 0.01623704, 0.01623303, 0.01621181,
       0.01619908, 0.01619218, 0.01618169, 0.01616813, 0.01615441,
       0.01613769, 0.01612569, 0.01609127, 0.01608051, 0.01606488,
       0.01605342, 0.01601542, 0.01600378, 0.0159909 , 0.01598511,
       0.01597465, 0.01597111, 0.01594019, 0.01593833, 0.01592567,
       0.01592559, 0.01592326, 0.01591412, 0.01588383, 0.01580828,
       0.01577081, 0.01575238, 0.01573619, 0.01572976, 0.01572463,
       0.01571544, 0.01568853, 0.01568788, 0.0156276 , 0.0155335 ,
       0.01552813, 0.01551286, 0.01551169, 0.0155008 , 0.01549527,
       0.01548694, 0.01547107, 0.01542627, 0.01541719, 0.01541673,
       0.01537991, 0.01535395, 0.01534628, 0.01532152, 0.01531945,
       0.01528668, 0.01526955, 0.0152673 , 0.01525957, 0.01524647,
       0.01523589, 0.01523018, 0.01516183, 0.01515407, 0.01514555,
       0.015122  , 0.01509092, 0.01507152, 0.01503686, 0.01499089,
       0.01498514, 0.01497624, 0.01497196, 0.01496111, 0.01492574,
       0.01492423, 0.01488356, 0.01487318, 0.0148725 , 0.01484371,
       0.01479802, 0.01474583, 0.014719  , 0.01470845, 0.01467155,
       0.0146449 , 0.01460885, 0.0146025 , 0.0145999 , 0.01459682,
       0.01458213, 0.01454482, 0.01454314, 0.01451374, 0.01451205,
       0.01449863, 0.01448859, 0.01448602, 0.01448327, 0.01448127,
       0.01445706, 0.01445499, 0.01444459, 0.01440722, 0.01435124,
       0.01433683, 0.01432857, 0.01430823, 0.01430715, 0.01429551,
       0.01429512, 0.01427863, 0.01427618, 0.01426682, 0.0142658 ,
       0.01426438, 0.01426058, 0.01421182, 0.0142017 , 0.01420126,
       0.01419849, 0.01417918, 0.01417739, 0.01417049, 0.01416892,
       0.01416264, 0.01414542, 0.01410483, 0.01410451, 0.01409259,
       0.01408876, 0.01408308, 0.01407129, 0.01405199, 0.01404307,
       0.014027  , 0.01402546, 0.01402516, 0.01401052, 0.0139996 ,
       0.01399539, 0.0139544 , 0.01391214, 0.01391209, 0.01390835,
       0.01389581, 0.01389261, 0.01388537, 0.01387862, 0.01387742,
       0.01385583, 0.01385332, 0.01385325, 0.01385231, 0.01384612,
       0.01384006, 0.0138303 , 0.01381997, 0.01381968, 0.01381563,
       0.01380857, 0.01379374, 0.01378763, 0.01378338, 0.01375897,
       0.01374664, 0.01373788, 0.01373085, 0.01373054, 0.01370686,
       0.01370504, 0.01368415, 0.01366565, 0.01365494, 0.01363912,
       0.01362417, 0.01360297, 0.01356894, 0.0135677 , 0.01355875,
       0.0135521 , 0.01354888, 0.01354833, 0.01354402, 0.01354092,
       0.01353594, 0.01352648, 0.01352525, 0.01351777, 0.01351391,
       0.01347948, 0.0134619 , 0.01346125, 0.01346116, 0.01343589,
       0.01342112, 0.01342111, 0.01339529, 0.01338935, 0.01337699,
       0.01336935, 0.01336927, 0.01336415, 0.01336333, 0.01335588,
       0.01334759, 0.0133341 , 0.01332073, 0.01330884, 0.01328763],
      dtype=float32)

Any idea?

Thanks,
Steve

**Comment by thusinh1969** [comment]
Found it. Use tf.keras.applications.resnet50.preprocess_input

Steve
